src/pages/playground/Decoder.tsx

```diff
@@ -0,0 +1,155 @@
+import { FSModule } from 'browserfs/dist/node/core/FS';
+import { Chapter, Variant } from 'js-slang/dist/types';
+import { decompressFromEncodedURIComponent } from 'lz-string';
+import { Dispatch } from 'react';
+import { AnyAction } from 'redux';
+import { getDefaultFilePath, getLanguageConfig } from 'src/commons/application/ApplicationTypes';
+import { overwriteFilesInWorkspace } from 'src/commons/fileSystem/utils';
+import {
+  showFullJSWarningOnUrlLoad,
+  showFulTSWarningOnUrlLoad,
+  showHTMLDisclaimer
+} from 'src/commons/utils/WarningDialogHelper';
+import {
+  addEditorTab,
+  removeEditorTabsForDirectory,
+  setFolderMode,
+  updateActiveEditorTabIndex
+} from 'src/commons/workspace/WorkspaceActions';
+import { WorkspaceLocation } from 'src/commons/workspace/WorkspaceTypes';
+import { playgroundConfigLanguage } from 'src/features/playground/PlaygroundActions';
+
+import { convertParamToBoolean, convertParamToInt } from '../../commons/utils/ParamParseHelper';
+import { IParsedQuery, parseQuery } from '../../commons/utils/QueryHelper';
+import { WORKSPACE_BASE_PATHS } from '../fileSystem/createInBrowserFileSystem';
+
+export type programConfig = {
+  isFolder: string | undefined;
+  tabs: string | undefined;
+  tabIdx: string | undefined;
+  chap: string | undefined;
+  variant: string | undefined;
+  ext: string | undefined;
+  exec: string | undefined;
+  files: string | undefined;
+  prgrm: string | undefined;
```

Comment:
Are they `| undefined` or just partial? Type-wise, they are the same. Semantically, they are different.

Or do you have a full config and use the `Partial` utility type?

Please refactor.